datajunction-server/datajunction_server/api/data.py

```diff
@@ -174,6 +174,22 @@ def get_data(  # pylint: disable=too-many-locals
     return result
 
 
+@router.get(
+    "/data/query/{query_id}",
+    response_model=QueryWithResults,
```

Comment:
What if data is no longer available for that query ID? Will it just return an empty list or do we need a new query results status?

Reply:
Let me test that... and maybe add a wrapper.

Reply:
Instead of a new results status I would just raise a `DJException` that has `http_status_code=HTTPStatus.NOT_FOUND` so it returns a 404.